go/host/rpc/enclaveclient/enclave_client_external.go

```diff
@@ -0,0 +1,240 @@
+package enclaveclient
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/rpc"
+	"github.com/obscuronet/go-obscuro/go/common/rpc/generated"
+	"github.com/obscuronet/go-obscuro/go/common/tracers"
+	"github.com/obscuronet/go-obscuro/go/config"
+	"github.com/obscuronet/go-obscuro/go/responses"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+	"google.golang.org/grpc"
+)
+
+type EnclaveExternalClient struct {
+	protoClient generated.EnclaveProtoClient
+	connection  *grpc.ClientConn
+	config      *config.HostConfig
+	logger      gethlog.Logger
+}
+
+func NewEnclaveExternalClient(
+	protoClient generated.EnclaveProtoClient,
+	connection *grpc.ClientConn,
+	config *config.HostConfig,
+	logger gethlog.Logger,
+) *EnclaveExternalClient {
+	return &EnclaveExternalClient{
+		protoClient: protoClient,
+		connection:  connection,
+		config:      config,
+		logger:      logger,
+	}
+}
+
+func (c *EnclaveExternalClient) SubmitTx(tx common.EncryptedTx) responses.RawTx {
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), c.config.EnclaveRPCTimeout)
+	defer cancel()
+
+	response, err := c.protoClient.SubmitTx(timeoutCtx, &generated.SubmitTxRequest{EncryptedTx: tx})
+	if err != nil {
```

Comment:
These 4 lines are duplicated on every call, might be good to wrap them in a standardised method.
```
	if err != nil {
		return responses.AsPlaintextError(err)
	}
	return *responses.ToEnclaveResponse(response.EncodedEnclaveResponse)
```

Reply:
Yeap, saw that too, will change that once I introduce the SystemError err.